chain/db/store.go

```diff
@@ -464,7 +372,10 @@ func (cs *ChainStore) persist(b *block.Block) error {
 		return err
 	}
 
-	cs.States = states
+	cs.States, err = NewStateDB(root, NewTrieStore(cs.GetDatabase()))
```

Comment:
If we get error here, the above persist functions are already executed, but cs.States will be empty. Will that cause a problem?

Reply:
hm, how about panic when returning error?